File: c/armstrong_numbers.c
#include "armstrong_numbers.h"

#include <math.h>
#include <stddef.h>
#include <stdio.h>
#include <string.h>

size_t len(int candidate);

size_t len(int candidate) {
  char s[256];
  sprintf(s, "%d", candidate);
  return strlen(s);
}

bool is_armstrong_number(int candidate) {
  size_t l = len(candidate);
  int sum = 0;
  int curr = candidate;
  while (curr > 0) {
    int d = curr % 10;
    sum += pow(d, l);
    if (sum > candidate)
      return false;
    curr /= 10;
  }
  return sum == candidate;
}

File: go/armstrong_numbers.go
package armstrong

import "math"

func IsNumber(n int) bool {
	ct := 0.0
	for m := n; m > 0; m /= 10 {
		ct++
	}
	sum := 0.0
	for m := n; m > 0; m /= 10 {
		sum += math.Pow(float64(m%10), ct)
	}
	return int(sum) == n
}
